pkg/abi/typecomponents.go

```diff
@@ -227,7 +227,7 @@ var (
 		},
 		jsonEncodingType: JSONEncodingTypeBytes,
 		encodeABIData:    encodeABIUnsignedInteger,
-		decodeABIData:    decodeABIUnsignedInt,
+		decodeABIData:    decodeABIAddress,
```

Comment:
Sorry @SamMayWork - I think I need to understand this better.
Maybe I need to work through this from base principals.

`decodeABIUnsignedInt` should be processing exactly 20 bytes.

The ABI specification states that `address` is processed identically to `uint256`.

So on the face of it, your change seems to sidestep a bug in `decodeABIUnsignedInt` by introducing a new function... but I don't understand why.

Reply:
This is a fair point - address is _just_ a `unit256` under the hood, so this fixed the problem seen on the BSC chain here, but if someone padded a normal `unit256` with non-zero bytes, we'd actually hit the same issue.

Taking the fixed code and moving it into the baseline `unit256` function...

Reply:
> This issue is causing golang to panic

Then just as important. A `uint256` should not panic the VM, including `ffffffffffffffffffffffffab0974bbed8afc5212e951c8498873319d02d025` (which to my knowledge is a perfectly valid `uint256`)

Reply:
So I'm expecting to see @SamMayWork :
- Processing of `ffffffffffffffffffffffffab0974bbed8afc5212e951c8498873319d02d025` as a `uint160` resulting in `0xab0974bbed8afc5212e951c8498873319d02d025` as a number
- Processing of `ffffffffffffffffffffffffab0974bbed8afc5212e951c8498873319d02d025` as a `uint8` resulting in `0x25` as a number
- Processing of `ffffffffffffffffffffffffab0974bbed8afc5212e951c8498873319d02d025` as a `uint256` processing `0xffffffffffffffffffffffffab0974bbed8afc5212e951c8498873319d02d025` as a number
- Processing of `ffffffffffffffffffffffffab0974bbed8afc5212e951c8498873319d02d025`  as an `address` processing `0xab0974bbed8afc5212e951c8498873319d02d025` with the semantic that `uint160` and `address` are equivalent at this layer of code

Reply:
Should be resolved by https://github.com/hyperledger/firefly-signer/pull/56/commits/711fd0f9bd4bfbe676189a25746cfe4e28ab3424